tools/autograd/templates/Functions.cpp

```diff
@@ -1974,6 +1974,27 @@ Tensor to_dense_backward(const Tensor& grad, const Tensor& input_) {
   return grad.sparse_mask(at::SparseTensorRef(input));
 }
 
+Tensor gather_sparse_backward(const Tensor& self, int64_t dim, const Tensor& index, const Tensor& grad){
+// special case scalar input and/or index
+    if (self.ndimension() == 0) return at::_sparse_coo_tensor_unsafe(at::empty({0,1}, index.options()), grad, self.sizes());
```

Comment:
That's a good catch, but what should the correct behavior be? The sparse gradient would be 1D (there's no way to make it scalar, because it has several elements). Is it ok to have 1D sparse gradient for scalar input? One way around would be to call .coalesce() and possibly reshape right away, that way scalar gradient may be preserved, but then, we are not calling .coalesce() for non-degenerate cases, and simply return sparse gradients with repeating indices.

Reply:
sorry, why would the sparse gradient need to be 1-D?  In the specific case, is it not correct to change:
```
at::_sparse_coo_tensor_unsafe(at::empty({0,1}, index.options()), grad, self.sizes());
```
to:
```
at::_sparse_coo_tensor_unsafe(at::empty({0,grad.numel()}, index.options()), grad, self.sizes());
```

Reply:
Cool! Did not know it works like this, I though indices.numel() and values.numel() should be the same.

Reply:
I usually have to look up the invariants each time to double-check, here is where I look:
https://github.com/pytorch/pytorch/blob/c3a23379ea80de93321a89ea63d939c6c66b9a0d/aten/src/ATen/SparseTensorImpl.h#L11-L15